index.php

```diff
@@ -38,14 +38,24 @@
 $languages = array();
 if ($handle = opendir('lang')) {
     while (false !== ($entry = readdir($handle))) {
-        if ($entry != "." && $entry != "..") {
-             array_push($languages, str_replace(".inc.php", "", $entry));
+        if ($entry != "." && $entry != ".." ) {
+            $entry_lang = str_replace(".inc.php", "", $entry);
+            # Only add language to possibilities if it is the default language or part of the allowed languages 
+            # empty $allowed_lang <=> all languages are allowed
+            if ($entry_lang == $lang || empty($allowed_lang) || in_array($entry_lang, $allowed_lang) ) {
+                array_push($languages, $entry_lang);
+            }
         }
     }
     closedir($handle);
 }
 $lang = detectLanguage($lang, $languages);
-require_once("lang/$lang.inc.php");
+if (file_exists("lang/$lang.inc.php")) {
```

Comment:
I don't like the idea of leaving the `require_once("lang/$lang.inc.php");` alone. This would leave a blank page, without any message at all for the site admin to investigate what's happening.

I would rather add a line to the error log when switching to english. This would be more helpful to investigate why the default language is not selected.